### ISSUE TITLE
Unusually high estimate of tokens

### ISSUE THREAD

**Issue by tomsib2001** [npiv/chatblade#17]
$ chatblade what is 1  + 1 --raw -t
what is 1 + 1

          tokens/costs
┏━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━┓
┃ Model   ┃ Tokens ┃     Price ┃
┡━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━┩
│ gpt-3.5 │ 4103   │ $0.008206 │
│ gpt-4   │ 4103   │ $0.123090 │
└─────────┴────────┴───────────┘
 * estimated costs do not include the tokens that may be returned

**Comment by npiv** [comment]
<img width="546" alt="image" src="https://user-images.githubusercontent.com/452020/226636333-a25c27dc-d084-49b5-99b1-2f5b99af2aa2.png">

Can't reproduce that. But it seems the token count is wrong it gives you `4103`. That is being done by the [tiktoken](https://github.com/openai/tiktoken) library. 

could you check what version of tiktoken you are running `pip list | grep tiktoken`. And try and update.

My local version is 0.3.2

**Comment by tomsib2001** [comment]
```shell
thomas@thomas-XPS-15-9520:~$ chatblade what is 1 + 1 --raw -t

what is 1 + 1

          tokens/costs
┏━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━┓
┃ Model   ┃ Tokens ┃     Price ┃
┡━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━┩
│ gpt-3.5 │ 4103   │ $0.008206 │
│ gpt-4   │ 4103   │ $0.123090 │
└─────────┴────────┴───────────┘
 * estimated costs do not include the tokens that may be returned
thomas@thomas-XPS-15-9520:~$
thomas@thomas-XPS-15-9520:~$ pip list | grep tiktoken
tiktoken                 0.3.2
thomas@thomas-XPS-15-9520:~$ pip install tiktoken -U
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: tiktoken in ./.local/lib/python3.10/site-packages (0.3.2)
Requirement already satisfied: regex>=2022.1.18 in ./.local/lib/python3.10/site-packages (from tiktoken) (2022.10.31)
Requirement already satisfied: requests>=2.26.0 in ./.local/lib/python3.10/site-packages (from tiktoken) (2.28.2)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./.local/lib/python3.10/site-packages (from requests>=2.26.0->tiktoken) (1.26.15)
Requirement already satisfied: charset-normalizer<4,>=2 in ./.local/lib/python3.10/site-packages (from requests>=2.26.0->tiktoken) (3.1.0)
Requirement already satisfied: idna<4,>=2.5 in ./.local/lib/python3.10/site-packages (from requests>=2.26.0->tiktoken) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in ./.local/lib/python3.10/site-packages (from requests>=2.26.0->tiktoken) (2022.12.7)
thomas@thomas-XPS-15-9520:~$ chatblade what is 1 + 1 --raw -t
what is 1 + 1

          tokens/costs
┏━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━┓
┃ Model   ┃ Tokens ┃     Price ┃
┡━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━┩
│ gpt-3.5 │ 4103   │ $0.008206 │
│ gpt-4   │ 4103   │ $0.123090 │
└─────────┴────────┴───────────┘
 * estimated costs do not include the tokens that may be returned
thomas@thomas-XPS-15-9520:~$ pip install chatblade -U
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: chatblade in ./.local/lib/python3.10/site-packages (0.0.2)
Requirement already satisfied: pyyaml in ./.local/lib/python3.10/site-packages (from chatblade) (6.0)
Requirement already satisfied: rich in ./.local/lib/python3.10/site-packages (from chatblade) (13.3.2)
Requirement already satisfied: tiktoken in ./.local/lib/python3.10/site-packages (from chatblade) (0.3.2)
Requirement already satisfied: platformdirs in /usr/lib/python3/dist-packages (from chatblade) (2.5.1)
Requirement already satisfied: openai>=0.27.2 in ./.local/lib/python3.10/site-packages (from chatblade) (0.27.2)
Requirement already satisfied: tqdm in ./.local/lib/python3.10/site-packages (from openai>=0.27.2->chatblade) (4.65.0)
Requirement already satisfied: aiohttp in ./.local/lib/python3.10/site-packages (from openai>=0.27.2->chatblade) (3.8.4)
Requirement already satisfied: requests>=2.20 in ./.local/lib/python3.10/site-packages (from openai>=0.27.2->chatblade) (2.28.2)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in ./.local/lib/python3.10/site-packages (from rich->chatblade) (2.14.0)
Requirement already satisfied: markdown-it-py<3.0.0,>=2.2.0 in ./.local/lib/python3.10/site-packages (from rich->chatblade) (2.2.0)
Requirement already satisfied: regex>=2022.1.18 in ./.local/lib/python3.10/site-packages (from tiktoken->chatblade) (2022.10.31)
Requirement already satisfied: mdurl~=0.1 in ./.local/lib/python3.10/site-packages (from markdown-it-py<3.0.0,>=2.2.0->rich->chatblade) (0.1.2)
Requirement already satisfied: idna<4,>=2.5 in ./.local/lib/python3.10/site-packages (from requests>=2.20->openai>=0.27.2->chatblade) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in ./.local/lib/python3.10/site-packages (from requests>=2.20->openai>=0.27.2->chatblade) (2022.12.7)
Requirement already satisfied: charset-normalizer<4,>=2 in ./.local/lib/python3.10/site-packages (from requests>=2.20->openai>=0.27.2->chatblade) (3.1.0)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./.local/lib/python3.10/site-packages (from requests>=2.20->openai>=0.27.2->chatblade) (1.26.15)
Requirement already satisfied: yarl<2.0,>=1.0 in ./.local/lib/python3.10/site-packages (from aiohttp->openai>=0.27.2->chatblade) (1.8.2)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in ./.local/lib/python3.10/site-packages (from aiohttp->openai>=0.27.2->chatblade) (4.0.2)
Requirement already satisfied: frozenlist>=1.1.1 in ./.local/lib/python3.10/site-packages (from aiohttp->openai>=0.27.2->chatblade) (1.3.3)
Requirement already satisfied: attrs>=17.3.0 in ./.local/lib/python3.10/site-packages (from aiohttp->openai>=0.27.2->chatblade) (22.2.0)
Requirement already satisfied: aiosignal>=1.1.2 in ./.local/lib/python3.10/site-packages (from aiohttp->openai>=0.27.2->chatblade) (1.3.1)
Requirement already satisfied: multidict<7.0,>=4.5 in ./.local/lib/python3.10/site-packages (from aiohttp->openai>=0.27.2->chatblade) (6.0.4)
thomas@thomas-XPS-15-9520:~$ chatblade what is 1 + 1 --raw -t
what is 1 + 1

          tokens/costs
┏━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━┓
┃ Model   ┃ Tokens ┃     Price ┃
┡━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━┩
│ gpt-3.5 │ 4103   │ $0.008206 │
│ gpt-4   │ 4103   │ $0.123090 │
└─────────┴────────┴───────────┘
 * estimated costs do not include the tokens that may be returned
 ```

**Comment by npiv** [comment]
Thanks for the confirmation. It should be fixed in the latest version

I just released 0.1.0 to pypi, or you can build from main branch what suits you best. Please reopen if it is still broken.
